docs/tutorials/python/mnist.md

```diff
@@ -57,7 +57,7 @@ data = mx.sym.flatten(data=data)
 ```
 One might wonder if we are discarding valuable information by flattening. That is indeed true and we'll cover this more when we talk about convolutional neural networks where we preserve the input shape. For now, we'll go ahead and work with flattened images.
 
-MLPs contains several fully connected layers. A fully connected layer or FC layer for short, is one where each neuron in the layer is connected to every neuron in its preceding layer. From a linear algebra perspective, an FC layer applies an [affine transform](https://en.wikipedia.org/wiki/Affine_transformation) to the *n x m* input matrix *X* and outputs a matrix *Y* of size *n x k*, where *k* is the number of neurons in the FC layer. *k* is also referred to as the hidden size. The output *Y* is computed according to the equation *Y = W X + b*. The FC layer has two learnable parameters, the *m x k* weight matrix *W* and the *m x 1* bias vector *b*.
+MLPs contains several fully connected layers. A fully connected layer or FC layer for short, is one where each neuron in the layer is connected to every neuron in its preceding layer. From a linear algebra perspective, an FC layer applies an [affine transform](https://en.wikipedia.org/wiki/Affine_transformation) to the *n x m* input matrix *X* and outputs a matrix *Y* of size *n x k*, where *k* is the number of neurons in the FC layer. *k* is also referred to as the hidden size. The output *Y* is computed according to the equation *Y = X W<sup>T</sup> + b*. The FC layer has two learnable parameters, the *k x m* weight matrix *W* and the *1 x k* bias vector *b*.
```

Comment:
"hidden size" is sounding a bit off. Is it only me? If not, probably "hidden layer size" or something like it may be better?

Another nitpick: bias vector dimension doesn't match in the number of rows with the output of X W<sup>T</sup>. Should we hint at broadcasting?

looks good to me otherwise.

Reply:
I had similar feeling wrt "hidden size" @pracheer. However Googling shows that "hidden size" is often used to refer to k and hidden layer dimension is used to refer to nxk (which is also output dimension). I also Googled 'hidden layer size' and this is also used to refer to the term k. I'd say it's a toss up.

WRT broadcasting, I searched MXNet's API docs for description of broadcasting (e.g. TF contains a broadcasting semantics page which includes basic explanation as well as linking to Numpy's documentation for deeper understanding (https://www.tensorflow.org/performance/xla/broadcasting). Pytorch has a similar page(http://pytorch.org/docs/master/notes/broadcasting.html) ). However I am not able to find any description of broadcasting on MXNet docs. Are you aware of such a page in MXNet? If not, I can add a link to numpy's broadcasting page, but I think we need to add a page for broadcasting to the doc.

Reply:
Even I'm not aware of an explicit page on broadcast apart from the information on [Symbol API](https://mxnet.incubator.apache.org/api/python/symbol.html) page. 

What you suggested sounds good to me: mentioning broadcasting (or linking numpy's page) for the time being and opening up a GitHub issue to add a page on broadcasting.

Reply:
I actually missed to see that the [`broadcast_to`](https://mxnet.incubator.apache.org/api/python/symbol.html#mxnet.symbol.broadcast_to) API doc in fact includes basic broadcasting semantics and does link to the numpy page. Perhaps the best option might be to have a paragraph at the top of Symbol API page and every API call with broadcasting linking to that paragraph.

Reply:
I'm not sure if that is the right place to put more info on additional concepts like broadcasting but can't think of anything better right now. @aaronmarkham any suggestions?